### DISCUSSION TITLE
PM.step() function

### DISCUSSION THREAD

**Discussion by mat194** [LAPKB/Pmetrics#231] [Q&A]
Hi everyone,

I have a question regarding the PM.step() function and its handling of categorical variables. Does the function account for the fact that some variables might be categorical and not purely numeric?

I attempted to replicate the results of the function to obtain coefficient estimates for my model implementation, but I noticed discrepancies between my results and those provided by the function. Additionally, it appears that the PM_data$new function does not accept non-numeric values for covariates. Am I missing something here?

For clarification, I am using the lm function to estimate the effects of my covariates on the parameters from the summarized data$cov results, not the entire dataset. Is this the correct approach? Should I be applying this to the EBE estimates instead?

Thank you very much for your assistance!

**Comment by mnneely** [comment]
Covariates must be numerical, otherwise, how would you incorporate them into equations? `$step()` is just a wrapper for the R [step](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/step) function. It uses the R [lm](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/lm) function. Since all covariates are numbers, `lm` handles them as continuous, even though the values are discrete, instead of creating pseudovariables as it would for factors. This might affect the statistical results.
